ext/libxml/libxml.c

```diff
@@ -378,9 +378,9 @@ php_libxml_input_buffer_create_filename(const char *URI, xmlCharEncoding enc)
 				const char buf[] = "Content-Type:";
 				if (Z_TYPE_P(header) == IS_STRING &&
 						!zend_binary_strncasecmp(Z_STRVAL_P(header), Z_STRLEN_P(header), buf, sizeof(buf)-1, sizeof(buf)-1)) {
-					char *needle = estrdup("charset=");
+					char needle[] = "charset=";
 					char *haystack = estrndup(Z_STRVAL_P(header), Z_STRLEN_P(header));
-					char *encoding = php_stristr(haystack, needle, Z_STRLEN_P(header), sizeof("charset=")-1);
+					char *encoding = php_stristr(haystack, needle, Z_STRLEN_P(header), strlen(needle));
```

Comment:
```suggestion
					const char *encoding = php_memnistr(haystack, "charset=", Z_STRLEN_P(header), sizeof("charset=") - 1);
```

Reply:
Any reason to prefer `sizeof() - 1` over `strlen()`? This should get optimized by the compiler anyway.

Reply:
https://stackoverflow.com/a/50238040 it seems like some of the time, it isn't done when compiler optimizations aren't enabled (e.g. MSVC for Windows without `/O2`, tiny C compiler) according to https://godbolt.org/

It seems like clang/gcc optimize it.

I'm not familiar with the reasons why it's done, only that it's widely done in this codebase.

Reply:
See also #5174 where this topic has been discussed; TL;DR: using `strlen()` should be fine nowadays.